### PR TITLE
Make CS pin on the screen optional

### PR DESCRIPTION
**Pull request by mmoskal**

### CODE REVIEW

File: inc/drivers/ST7735.h [hunk 1]
```diff
@@ -58,6 +58,11 @@ class ST7735 : public CodalComponent
     // and doesn't support 12 bit color
     bool double16;
 
+    void beginCS() { if (cs) cs->setDigitalValue(0); }
+    void endCS() { if (cs) cs->setDigitalValue(1); }
+    void setCommand() { dc->setDigitalValue(0); }
```
Comment (jamesadevine) [inline inc/drivers/ST7735.h]: dc should also be if guarded just in case?

File: source/drivers/ST7735.cpp [hunk 2]
```diff
@@ -65,7 +65,7 @@
 namespace codal
 {
 
-ST7735::ST7735(ScreenIO &io, Pin &cs, Pin &dc) : io(io), cs(cs), dc(dc), work(NULL)
+ST7735::ST7735(ScreenIO &io, Pin &cs, Pin &dc) : io(io), cs(&cs), dc(&dc), work(NULL)
```
Comment (jamesadevine) [inline source/drivers/ST7735.cpp]: I think at this point, cs and dc should just be pointers... references are only useful for mandatory parameters.

Comment (jamesadevine) [reply source/drivers/ST7735.cpp]: Default parameters as null?

Up to you on both points.

File: inc/drivers/ST7735.h (hunk 1 above)
Comment (mmoskal) [reply inc/drivers/ST7735.h]: no, the screen just won't work without dc, but it works fine without cs

File: source/drivers/ST7735.cpp (hunk 2 above)
Comment (mmoskal) [reply source/drivers/ST7735.cpp]: yeah, but I would have to change stuff in pxt then...

**Pull request merged**